database/database.go

```diff
@@ -59,3 +60,65 @@ func GetBlocksAfter(blockHash Hash, dataDir string) ([]Block, error) {
 
 	return blocks, nil
 }
+
+// GetBlockByHeightOrHash returns the requested block by hash or height.
+// Add a map cache of height and hash to State struct
+// and the corresponding functions to update and read this info
+// The value of the maps corresponds to the position on the file corresponding to the block
+
+//
+//  type State struct {
```

Comment:
These comments you could remove and just reference the dev to see the State struct.